bldsys/cmake/sample_helper.cmake

```diff
@@ -253,4 +266,38 @@ endif()
             set_property(TARGET ${bare_name}-${extension} PROPERTY FOLDER "HLSL_Shaders")
         endforeach()
     endif()
+
+    # Slang shader compilation
+    if(Vulkan_slang_EXECUTABLE AND DEFINED SHADERS_SLANG)
+        foreach(SHADER_FILE_SLANG ${TARGET_SHADERS_SLANG})
+            get_filename_component(SLANG_SPV_FILE ${SHADER_FILE_SLANG} NAME_WLE)
+            get_filename_component(bare_name ${SLANG_SPV_FILE} NAME_WLE)
+            get_filename_component(extension ${SLANG_SPV_FILE} LAST_EXT)
+            get_filename_component(directory ${SHADER_FILE_SLANG} DIRECTORY)
+            string(REGEX REPLACE "[.]+" "" extension ${extension})
+            set(OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/shader-slang-spv")
+            set(OUTPUT_FILE ${OUTPUT_DIR}/${bare_name}.${extension}.spv)
+            file(MAKE_DIRECTORY ${OUTPUT_DIR})
+
+            set(SLANG_PROFILE "glsl_460")
+            set(SLANG_ENTRY_POINT "main")
+            add_custom_command( PRE_BUILD
+                    OUTPUT ${OUTPUT_FILE}
+                    COMMAND ${Vulkan_slang_EXECUTABLE} ${SHADER_FILE_SLANG} -profile ${SLANG_PROFILE} -target spirv -o ${OUTPUT_FILE} -entry ${SLANG_ENTRY_POINT}
+                    COMMAND ${CMAKE_COMMAND} -E copy ${OUTPUT_FILE} ${directory}
```

Comment:
You can have multiple entries in one spirv file.  So, instead of having 1 vert shader and 1 frag shader file, you can join them into one slang file and give each an entry point.
This is beneficial especially for ray tracing:
slangc shaders.slang -target spirv -profile spirv_1_4 -emit-spirv-directly -fvk-use-entrypoint-name  -entry vertMain -entry fragMain -o slang.spv
is an example of how to achieve that.

Reply:
Yes, I am aware of that (and use it in my own projects). But that would've required additional work to our framework, so I stuck with separate shader files.